From http://trac.wxwidgets.org/ticket/12575 :

  gcc refuses to compile trunk/wxGTK in C++0x mode. The culprit is that
  in the .xbm files char-arrays are initialised with values >127. This
  does not fly if char==signed char, e.g:

  ../src/gtk/bdiag.xbm:6:50: error: narrowing conversion of ‘128’ from ‘int’ to ‘char’ inside { } [-fpermissive]

Reviewed-and-rediffed-by: Lucas C. Villa Real <lucasvr@gobolinux.org>

Index: a/src/gtk/dcclient.cpp
===================================================================
--- a/src/gtk/dcclient.cpp.orig	2017-01-10 09:51:56.610044286 -0200
+++ b/src/gtk/dcclient.cpp	2017-01-10 09:53:25.031045945 -0200
@@ -428,12 +428,12 @@ void wxWindowDC::SetUpDC()
     if (!hatch_bitmap)
     {
         hatch_bitmap    = hatches;
-        hatch_bitmap[0] = gdk_bitmap_create_from_data( (GdkWindow *) NULL, bdiag_bits, bdiag_width, bdiag_height );
-        hatch_bitmap[1] = gdk_bitmap_create_from_data( (GdkWindow *) NULL, cdiag_bits, cdiag_width, cdiag_height );
-        hatch_bitmap[2] = gdk_bitmap_create_from_data( (GdkWindow *) NULL, fdiag_bits, fdiag_width, fdiag_height );
-        hatch_bitmap[3] = gdk_bitmap_create_from_data( (GdkWindow *) NULL, cross_bits, cross_width, cross_height );
-        hatch_bitmap[4] = gdk_bitmap_create_from_data( (GdkWindow *) NULL, horiz_bits, horiz_width, horiz_height );
-        hatch_bitmap[5] = gdk_bitmap_create_from_data( (GdkWindow *) NULL, verti_bits, verti_width, verti_height );
+        hatch_bitmap[0] = gdk_bitmap_create_from_data( (GdkWindow *) NULL, reinterpret_cast<const char*>(bdiag_bits), bdiag_width, bdiag_height );
+        hatch_bitmap[1] = gdk_bitmap_create_from_data( (GdkWindow *) NULL, reinterpret_cast<const char*>(cdiag_bits), cdiag_width, cdiag_height );
+        hatch_bitmap[2] = gdk_bitmap_create_from_data( (GdkWindow *) NULL, reinterpret_cast<const char*>(fdiag_bits), fdiag_width, fdiag_height );
+        hatch_bitmap[3] = gdk_bitmap_create_from_data( (GdkWindow *) NULL, reinterpret_cast<const char*>(cross_bits), cross_width, cross_height );
+        hatch_bitmap[4] = gdk_bitmap_create_from_data( (GdkWindow *) NULL, reinterpret_cast<const char*>(horiz_bits), horiz_width, horiz_height );
+        hatch_bitmap[5] = gdk_bitmap_create_from_data( (GdkWindow *) NULL, reinterpret_cast<const char*>(verti_bits), verti_width, verti_height );
     }
 }
 
Index: a/src/gtk/bdiag.xbm
===================================================================
--- a/src/gtk/bdiag.xbm	(revision 65839)
+++ b/src/gtk/bdiag.xbm	(working copy)
@@ -1,6 +1,6 @@
 #define bdiag_width 16
 #define bdiag_height 16
-static char bdiag_bits[] = {
+static unsigned char bdiag_bits[] = {
    0x80, 0x80, 0x40, 0x40, 0x20, 0x20, 0x10, 0x10, 0x08, 0x08, 0x04, 0x04,
    0x02, 0x02, 0x01, 0x01, 0x80, 0x80, 0x40, 0x40, 0x20, 0x20, 0x10, 0x10,
    0x08, 0x08, 0x04, 0x04, 0x02, 0x02, 0x01, 0x01};
Index: a/src/gtk/cdiag.xbm
===================================================================
--- a/src/gtk/cdiag.xbm	(revision 65839)
+++ b/src/gtk/cdiag.xbm	(working copy)
@@ -1,6 +1,6 @@
 #define cdiag_width 16
 #define cdiag_height 16
-static char cdiag_bits[] = {
+static unsigned char cdiag_bits[] = {
    0x81, 0x81, 0x42, 0x42, 0x24, 0x24, 0x18, 0x18, 0x18, 0x18, 0x24, 0x24,
    0x42, 0x42, 0x81, 0x81, 0x81, 0x81, 0x42, 0x42, 0x24, 0x24, 0x18, 0x18,
    0x18, 0x18, 0x24, 0x24, 0x42, 0x42, 0x81, 0x81};
Index: a/src/gtk/fdiag.xbm
===================================================================
--- a/src/gtk/fdiag.xbm	(revision 65839)
+++ b/src/gtk/fdiag.xbm	(working copy)
@@ -1,6 +1,6 @@
 #define fdiag_width 16
 #define fdiag_height 16
-static char fdiag_bits[] = {
+static unsigned char fdiag_bits[] = {
    0x01, 0x01, 0x02, 0x02, 0x04, 0x04, 0x08, 0x08, 0x10, 0x10, 0x20, 0x20,
    0x40, 0x40, 0x80, 0x80, 0x01, 0x01, 0x02, 0x02, 0x04, 0x04, 0x08, 0x08,
    0x10, 0x10, 0x20, 0x20, 0x40, 0x40, 0x80, 0x80};
Index: a/src/gtk/horiz.xbm
===================================================================
--- a/src/gtk/horiz.xbm	(revision 65839)
+++ b/src/gtk/horiz.xbm	(working copy)
@@ -1,6 +1,6 @@
 #define horiz_width 15
 #define horiz_height 15
-static char horiz_bits[] = {
+static unsigned char horiz_bits[] = {
    0x00, 0x00, 0x00, 0x00, 0xff, 0x7f, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
    0x00, 0x00, 0xff, 0x7f, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
    0xff, 0x7f, 0x00, 0x00, 0x00, 0x00};
Index: a/src/gtk/cross.xbm
===================================================================
--- a/src/gtk/cross.xbm	(revision 65839)
+++ b/src/gtk/cross.xbm	(working copy)
@@ -1,6 +1,6 @@
 #define cross_width 15
 #define cross_height 15
-static char cross_bits[] = {
+static unsigned char cross_bits[] = {
    0x84, 0x10, 0x84, 0x10, 0xff, 0x7f, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10,
    0x84, 0x10, 0xff, 0x7f, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10,
    0xff, 0x7f, 0x84, 0x10, 0x84, 0x10};
Index: a/src/gtk/verti.xbm
===================================================================
--- a/src/gtk/verti.xbm	(revision 65839)
+++ b/src/gtk/verti.xbm	(working copy)
@@ -1,6 +1,6 @@
 #define verti_width 15
 #define verti_height 15
-static char verti_bits[] = {
+static unsigned char verti_bits[] = {
    0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10,
    0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10, 0x84, 0x10,
    0x84, 0x10, 0x84, 0x10, 0x84, 0x10};
